pkg/products/cloudresources/reconciler.go

```diff
@@ -204,7 +204,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 
-	phase, err = r.newAlertsReconciler(r.log, r.installation.Spec.Type).ReconcileAlerts(ctx, client)
+	alertsReconciler, err := r.newAlertsReconciler(ctx, client, r.log, r.installation.Spec.Type, r.installation.Namespace)
+	if err != nil {
+		events.HandleError(r.recorder, installation, phase, "Failed to get new alerts reconciler", err)
+		r.log.Error("Error getting cloud resources alerts reconciler", err)
+		return phase, err
```

Comment:
hey, I'm a bit confused here. We are returning phase, err here but the output of newAlertsReconciler is the alertReconciler. Should the returning phase in this block be phaseFailed rather that phase returned from previous func